core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1347,59 +1348,47 @@ class KafkaApis(val requestChannel: RequestChannel,
 
   def handleCreateTopicsRequest(request: RequestChannel.Request) {
     val createTopicsRequest = request.body[CreateTopicsRequest]
+    val topicResults = mutable.ArrayBuffer[(String, ApiError)]()
 
     def sendResponseCallback(results: Map[String, ApiError]): Unit = {
       def createResponse(requestThrottleMs: Int): AbstractResponse = {
-        val responseBody = new CreateTopicsResponse(requestThrottleMs, results.asJava)
-        trace(s"Sending create topics response $responseBody for correlation id ${request.header.correlationId} to client ${request.header.clientId}.")
+        val responseData = new CreateTopicsResponseData().
+          setThrottleTimeMs(requestThrottleMs)
+        topicResults ++= results.toList
+        topicResults.foreach {
+          case (topicName, error) => responseData.topics().add(
+            new CreatableTopicResult().
+              setName(topicName).
+              setErrorCode(error.error().code()).
+              setErrorMessage(error.message()))
+        }
+        val responseBody = new CreateTopicsResponse(responseData)
+        trace(s"Sending create topics response $responseData for correlation id " +
+          "${request.header.correlationId} to client ${request.header.clientId}.")
         responseBody
       }
       sendResponseMaybeThrottle(request, createResponse)
     }
 
     if (!controller.isActive) {
-      val results = createTopicsRequest.topics.asScala.map { case (topic, _) =>
-        (topic, new ApiError(Errors.NOT_CONTROLLER, null))
+      createTopicsRequest.data.topics.asScala.foreach { case topic =>
+        topicResults += ((topic.name(), new ApiError(Errors.NOT_CONTROLLER, null)))
       }
-      sendResponseCallback(results)
+      sendResponseCallback(Map.empty)
```

Comment:
It is more for consistency with the code in the "else" block.

There might be a better way to structure, this though.  I have to say I would prefer just to have a map from topic names to results rather than dealing with all these lists.  But I didn't want to change AdminManager#createTopics.

Reply:
I have changed this in the latest version -- take a look.